crypto/ec/ecp_secp256k1.c

```diff
@@ -0,0 +1,2160 @@
+/*
+ * Copyright 2011-2024 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+/*
+ * The main parts of the code were extracted from
+ * https://github.com/llamasoft/secp256k1_fast_unsafe/ which was itself
+ * adapted from https://github.com/bitcoin-core/secp256k1 and adapted by
```

Comment:
If this code is taken from other authors, does it require the express permission of all authors in order to be submitted into OpenSSL?

Reply:
Yes.

Reply:
Isn't it what the original licence is supposed to cover ?

Reply:
No, unfortunately not.

Reply:
The low-level code was extracted from the [Bitcoin's repository](https://github.com/bitcoin-core/secp256k1/) , and this code was written ~10 years ago. Doesn't their licence mean that the code can be reused elsewhere without the burden of the paperwork ?
 
I copied the copyright text in all the files I submitted here, which I think, triggered the comments above, but the `COPYING.md` file explicitely says:
`
Copyright (c) 2013 Pieter Wuille

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in
all copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
THE SOFTWARE.
`

Maybe the CLA should be filled by the Bitcoin Fundation ? Or any other company maintaining this repository ?
I prefer waiting for this conversation to be resolved before working on solving the others

Reply:
The CLA would have to be filled by all the people which made non-trivial changes in the source files that you're taking from the Bitcoin repo including the original author.

The permissive license is nice but it does not overcome the need to have the CLA filled with us by all the authors.